src/screens/game/SubmitScoreDialog.tsx

```diff
@@ -37,7 +38,7 @@ const SubmitScoreDialog: React.FC<SubmitScoreDialogProps> = ({
     } else {
       setLoading(true);
 
-      await onSubmit(username);
+      await onSubmit(removeWhitespaces(username));
```

Comment:
Add this code to the text field onChange, so the users see their spaces get removed.